### ISSUE TITLE
[BUG]: Removal of POS Awesome corrupts ERPnext when upgraded to V15

### ISSUE THREAD

**Issue by blewsky** [yrestom/POS-Awesome#437]
I removed POS Awesome (v14) before upgrading to v15. Now my installation is broken and I cannot fix it since I do not have full bench/DB access on Frappe Cloud to fix this manually. It took me two days to try to fix this and I was not successful. Unfortunately, this issue blocks our operations now. There are some doctypes (e.g. sales order) that have POS fields added. I cannot remove or modify them since they are not custom but part of the system. E.g. the sales order links to POS-Awesome doctypes. When we want to submit a sales order, we get the following error message:
<img width="744" alt="image" src="https://github.com/yrestom/POS-Awesome/assets/87971956/5024dfe4-1db7-4c61-8c07-9f816afaa50a">
I took quite a while to debug this. Due to the POS-A app, the Sales Order links to POS Offer Detail and POS Coupon Detail. 
How can POS Awesome on Frappe Cloud be deinstalled/removed correctly/fully so we can upgrade and use v15?
When will POS-Awesome be available in v15? Probably not really likely that we can wait until the v15 will be available on the FC market place. 

### Versions
- Frappe: v15.7.0
- Erpnext: v15.8.0
- POS Awesome: v14?? (not visible anymore)

**Comment by blewsky** [comment]
It seems that the removal of the app is not correctly set. 
The custom fields in ERPnext remain after the deinstallation which causes problems (see above).
In my understanding Frappe does not handle the removal of custom fields, currently. This should be done by the app. See thread [here](https://github.com/frappe/frappe/pull/10871#issuecomment-1186897507).
We are still stuck and cannot use our ERPnext installation anymore. Also a bit frustrating that on Frappe Cloud there isn't the option for a free trial anymore (as it used to be), so I cannot reinstall POS Awesome - even if we don't use it, just to be able to use the core functionalities (like Sales Order). @yrestom Any suggestions how to resolve this?

**Comment by yrestom** [comment]
> It seems that the removal of the app is not correctly set. The custom fields in ERPnext remain after the deinstallation which causes problems (see above). In my understanding Frappe does not handle the removal of custom fields, currently. This should be done by the app. See thread [here](https://github.com/frappe/frappe/pull/10871#issuecomment-1186897507). We are still stuck and cannot use our ERPnext installation anymore. Also a bit frustrating that on Frappe Cloud there isn't the option for a free trial anymore (as it used to be), so I cannot reinstall POS Awesome - even if we don't use it, just to be able to use the core functionalities (like Sales Order). @yrestom Any suggestions how to resolve this?

Please get in touch with me by email to help you

**Comment by blewsky** [comment]
Thank you for your offer. After several hours of debugging yesterday, I finally managed to remove the broken POS Awesome installation.
I setup ssh access on Frappe Cloud, logged into the machine and ran on  `bench console`
` frappe.db.delete("Custom Field", filters={"fieldname": ["like", "posa_%"]})
frappe.db.commit()`
And then finally
` bench --site {site} migrate`
I hope this won't create problems down the road but it was the only option that I could get it fixed.
I resolved my specific issue but the general issue on a proper uninstall remains IMO. 
(See related bug report on frappe/frappe: https://github.com/frappe/frappe/issues/24108)
